core/ecschema-metadata/src/Metadata/Unit.ts

```diff
@@ -48,6 +48,19 @@ export class Unit extends SchemaItem {
   public get offset(): number { return this._offset; }
   public get denominator(): number { return this._denominator; }
 
+  /**
+   * Check if units phenomena's SchemaItemKey have the same names
+   * @internal
+   */
+  public static async areCompatiblePhenomenons(unitA: Unit, unitB: Unit): Promise<boolean> {
+    const unitAPhenomenon = await unitA.phenomenon;
+    const unitBPhenomenon = await unitB.phenomenon;
+
+    if (!unitAPhenomenon || !unitBPhenomenon || unitAPhenomenon.key.name !== unitBPhenomenon.key.name)

```

Comment:
This is actually wrong and I missed it until I saw the test schemas.  The schema and item name must match for them to be equal.  You're just testing the name of the item.  Add a test for this